modules/fbs/includes/fbs.availability.inc

```diff
@@ -94,11 +141,7 @@ function fbs_availability_holdings($provider_ids) {
         $result_holding['placement'][] = $material_description[0]->materialGroupName;
       }
 
-      module_load_include('client.inc', 'opensearch');
-      $entity_id = ding_provider_build_entity_id($provider_ids[0]);
-      $suffix_type = variable_get('fbs_holdings_suffix_type', 'shelf_mark');
-
-      if ($suffix_type == 'simple' && $entity = ding_entity_load($entity_id)) {
```

Comment:
Moving this out of the holdings loop seems to be a key change here.

Since `$provider_ids[0]` does not change for each iteration we can move all entity loading out of the loop.

Reply:
Not sure that I follow you here?